### PR TITLE
SE-3540 [LX-1653] Fix create_transcripts_xml when resource_fs is WrapFS

### PR DESCRIPTION
**Pull request by swalladge**
`resource_fs` can be set to a WrapFS instance in some cases,
notably in override_export_fs, which is used when exporting content via
the olx_rest_api.

This bug was introduced in https://github.com/edx/edx-val/pull/258,
which in turn fixed other bugs.
So this commit preserves the behaviour from that PR,
while ensuring that the original static_file_dir logic is used when
a WrapFS is passed (ie. no _sub_dir attr).

An alternate option would be to add AttributeError to an except clause,
but it feels safer to be more specific and proactive in this case.

**JIRA tickets**: [OSPR-5109](https://openedx.atlassian.net/browse/OSPR-5109)

**Merge deadline**: ASAP; this is breaking exporting videos, which is used by our clients.

**Test instructions**:

- ensure that the behaviour from https://github.com/edx/edx-val/pull/258 is retained. @DawoudSheraz, you may wish to check that this PR doesn't undo any of the fixes you did there. Please also check my comments for accuracy. :)  (I don't know how to test the original PR)
- use the olx export rest api to export a video (test with a video with and without transcripts)
- verify that the olx is successfully exported

**Reviewers**:

- [x] @gabor-boros
- [ ] edX reviewer TBD

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @swalladge! I've created [OSPR-5109](https://openedx.atlassian.net/browse/OSPR-5109) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by gabor-boros** [comment]
👍  🎉 

- [x] I tested this: checked if the changes fixes the issues for our client
- [x] I read through the code
- I checked for accessibility issues NA
- Includes documentation NA
- I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository. NA

**Comment by swalladge** [comment]
@natabene this is ready for edX review.  @DawoudSheraz may wish to look at this too since it relates to https://github.com/edx/edx-val/pull/258

**Comment by natabene** [comment]
@swalladge Thank you for your contribution. @DawoudSheraz Would you like to review this?

File: edxval/api.py
Comment (DawoudSheraz) [inline edxval/api.py:967]: A clarification of how this can be different will be helpful for future devs.

File: edxval/tests/test_api.py
Comment (DawoudSheraz) [inline edxval/tests/test_api.py:1152]: No need for this as the unit test docstring is already explaining this.

File: edxval/tests/test_api.py
Comment (DawoudSheraz) [inline edxval/tests/test_api.py:1193]: nit: move `,` after wrapfs and mention olx_rest_api after edx-platform to be more specific

File: edxval/tests/test_api.py
Comment (DawoudSheraz) [inline edxval/tests/test_api.py:1197]: These lines can be converted into a helper function, like _setup_wrap_fs() to avoid code duplication.

**Review by DawoudSheraz — COMMENTED** [review]
Good overall. Just need to address a few nits before merging.

**Comment by swalladge** [comment]
@DawoudSheraz thanks!  I addressed your comments. Ready for another pass. :smile:

**Review by DawoudSheraz — APPROVED** [review]

**Comment by DawoudSheraz** [comment]
@swalladge PR approved. If it is not waiting for another approval on your end, I will merge it. Please let me know. Thanks for your contribution.

**Comment by swalladge** [comment]
@DawoudSheraz thanks!  We have completed our internal review process (@gabor-boros [approved above](https://github.com/edx/edx-val/pull/267#issuecomment-722276449)), so will be great to get this merged! :smile:

**Comment by edx-webhook** [comment]
@swalladge 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-val&entry.867055334=https://github.com/edx/edx-val/pull/267&entry.1484655318=https://github.com/swalladge&entry.752974735=2020-11-05+03:37&entry.1917517419=2020-11-10+06:01&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by DawoudSheraz** [comment]
PR has been merged and a release tag https://github.com/edx/edx-val/releases/tag/1.4.4 has been created.

**Comment by swalladge** [comment]
Thanks @DawoudSheraz !  Could you let us know when this has been deployed to edX staging and production please? :smile:

**Comment by DawoudSheraz** [comment]
@swalladge Hi. This has reached production after the requirements update in https://github.com/edx/edx-platform/pull/25554. Thank You

**Comment by swalladge** [comment]
@DawoudSheraz thanks, that's good to know. :+1:

**Pull request merged**
